### PR TITLE
New starter spells

### PR DESCRIPTION
**Pull request by Rfpnj**
### Things to include in a large Pull Request:

- What is the purpose of the pull request?
Expands the Character Customization by set starting spell based on the month of birth
- Give a brief description of what you changed or added.
^
- Are any new graphical assets required?
No
- Has this change been tested? If so, mention the version number that the test was based on.
Yes, 2.12
- So we have a better idea of who you are, what is your Discord Handle?

- If you want quick feedback, you can use @Innoxia, or jump on the Lilith's Throne discord and send Innoxia a PM.

### CODE REVIEW

File: src/com/lilithsthrone/game/dialogue/story/CharacterCreation.java [hunk 1]
```diff
@@ -1797,9 +1797,19 @@ public void effects() {
 						Main.game.getPlayer().setMoney(5000);
 						Main.game.getPlayer().equipMainWeaponFromNowhere(AbstractWeaponType.generateWeapon(WeaponType.MELEE_CHAOS_RARE, DamageType.FIRE));
 						
-						AbstractItem spellBook = AbstractItemType.generateItem(ItemType.getSpellBookType(Spell.ICE_SHARD));
+						if (Main.game.getPlayer().getBirthMonth().getValue() == 1||Main.game.getPlayer().getBirthMonth().getValue() == 5||Main.game.getPlayer().getBirthMonth().getValue() == 9) {
```
Comment (CognitiveMist) [inline src/com/lilithsthrone/game/dialogue/story/CharacterCreation.java]: Suggestion for readability (at least, imo):

```suggestion
						if (Main.game.getPlayer().getBirthMonth().getValue() % 4 == 1) {
```

And then `% 4 == 2`, `% 4 == 3` and `% 4 == 0` for the others.

**Comment by Coldknife2** [comment]
Hmmm.. Not a fan

What I see is :  
```java
public void effects() {
                        stuff
                        if (Main.game.getPlayer().getBirthMonth().getValue() == 1||Main.game.getPlayer().getBirthMonth().getValue() == 5||Main.game.getPlayer().getBirthMonth().getValue() == 9) {
                            stuff
                        } else if (Main.game.getPlayer().getBirthMonth().getValue() == 2||Main.game.getPlayer().getBirthMonth().getValue() == 6||Main.game.getPlayer().getBirthMonth().getValue() == 10) {
                            stuff
                        } else if (Main.game.getPlayer().getBirthMonth().getValue() == 3||Main.game.getPlayer().getBirthMonth().getValue() == 7||Main.game.getPlayer().getBirthMonth().getValue() == 11) {
                            stuff
                        } else if (Main.game.getPlayer().getBirthMonth().getValue() == 4||Main.game.getPlayer().getBirthMonth().getValue() == 8||Main.game.getPlayer().getBirthMonth().getValue() == 12) {
                            stuff
                        };
                    }
```

When you could write : 
```java
switch(Main.game.getPlayer().getBirthMonth().getValue() % 4)
{
    case 1:
    stuff;
    break;

    case 2:
    stuff;
    break

    case 3:
    stuff;
    break

    case 4:
    stuff;
    break
}
```

File: src/com/lilithsthrone/game/dialogue/story/PrologueDialogue.java [hunk 2]
```diff
@@ -697,9 +697,20 @@ public Response getResponse(int responseTab, int index) {
...
+						} else if (Main.game.getPlayer().getBirthMonth().getValue() == 3||Main.game.getPlayer().getBirthMonth().getValue() == 7||Main.game.getPlayer().getBirthMonth().getValue() == 11) {
+							AbstractItem spellBook = AbstractItemType.generateItem(ItemType.getSpellBookType(Spell.ICE_SHARD));
+						Main.game.getWorlds().get(WorldType.LILAYAS_HOUSE_FIRST_FLOOR).getCell(PlaceType.LILAYA_HOME_ROOM_PLAYER).getInventory().addItem(spellBook);
+						} else if (Main.game.getPlayer().getBirthMonth().getValue() == 4||Main.game.getPlayer().getBirthMonth().getValue() == 8||Main.game.getPlayer().getBirthMonth().getValue() == 12) {
+							AbstractItem spellBook = AbstractItemType.generateItem(ItemType.getSpellBookType(Spell.FIREBALL));
+						Main.game.getWorlds().get(WorldType.LILAYAS_HOUSE_FIRST_FLOOR).getCell(PlaceType.LILAYA_HOME_ROOM_PLAYER).getInventory().addItem(spellBook);
 						Main.game.getTextEndStringBuilder().append("<p style='text-align:center;'>[style.boldExcellent("+spellBook.getName()+")] added to your room's storage!</p>");
+						};
```
Comment (Coldknife2) [inline src/com/lilithsthrone/game/dialogue/story/PrologueDialogue.java]: Hmmm.. Not a fan

What I see is :  
```java
public void effects() {
                        stuff
                        if (Main.game.getPlayer().getBirthMonth().getValue() == 1||Main.game.getPlayer().getBirthMonth().getValue() == 5||Main.game.getPlayer().getBirthMonth().getValue() == 9) {
                            stuff
                        } else if (Main.game.getPlayer().getBirthMonth().getValue() == 2||Main.game.getPlayer().getBirthMonth().getValue() == 6||Main.game.getPlayer().getBirthMonth().getValue() == 10) {
                            stuff
                        } else if (Main.game.getPlayer().getBirthMonth().getValue() == 3||Main.game.getPlayer().getBirthMonth().getValue() == 7||Main.game.getPlayer().getBirthMonth().getValue() == 11) {
                            stuff
                        } else if (Main.game.getPlayer().getBirthMonth().getValue() == 4||Main.game.getPlayer().getBirthMonth().getValue() == 8||Main.game.getPlayer().getBirthMonth().getValue() == 12) {
                            stuff
                        };
                    }
```

When you could write : 
```java
switch(Main.game.getPlayer().getBirthMonth().getValue() % 4)
{
    case 1:
    stuff;
    break;

    case 2:
    stuff;
    break

    case 3:
    stuff;
    break

    case 4:
    stuff;
    break
}
```

File: src/com/lilithsthrone/game/dialogue/story/CharacterCreation.java (hunk 1 above)
Comment (Coldknife2) [reply src/com/lilithsthrone/game/dialogue/story/CharacterCreation.java]: +1

**Comment by Innoxia** [comment]
Thank you for this! Sorry it took so long to merge!

**Pull request merged**